web-client/public/js/graph.js

```diff
@@ -91,6 +92,8 @@ export var drawGraph = function (network, sliderController, nodeColoring) {
         document.getElementById("edge-weight-normalization-factor-menu").setAttribute("placeholder", "");
     } else {
         var maxWeight = d3.max(allWeights);
+        grnState.resetNormalizationMax = maxWeight;
+        grnState.normalizationMax = maxWeight;
```

Comment:
Note that it was this assignment that triggered the shift of doing the entire `network` and loading/importing sequence first. In order to follow our new structure properly, _there should be no changes to the model outside of controller code_. When I first saw this on Tuesday, recall that we tracked down the origin of `maxWeight` to a calculation based purely on the network's weights (line 60/61; open up the diff to see it or look at your code directly).

This was the plan that we mapped out on Tuesday:
- [x] `grnState` now holds the current `network` (This has been done)
- [x] All loading and importing code now follows our new MVC flow: the triggers live in `setupHandlers`, and this code assigns the new `network` to `grnState`. Upon assigning `network`, `updateApp` refreshes the view (Also done)
- [x] The setting of the `network` is expected to trigger a cascade of related model calculations—_this is one of them_. To help you with implementation, the `network` property was implemented as a getter/setter, so that additional code can be executed when a new `network` is assigned (Also done)
- [ ] _Now this is what you should do:_ Expand the `network` setter code so that, when the network is assigned, _the calculation that leads to `maxWeight` is performed in `grnState`_. This will lead to the assignment of `resetNormalizationMax` and `normalizationMax`. `updateApp` then proceeds as it currently does, and if all goes well, the view is updated accordingly.

Remember, we are all about `grnState` -> `updateApp` -> controller code in `setupHandlers` -> `grnState`. _Do not digress from this flow unless there is a really good reason to do so_. (we have one so far—I marked it with a comment explaining the exception)

I hope this spells it out sufficiently for you. Let me know if you need further clarification.